File: Rust/src/search_insert_position.rs
pub fn search_insert(nums: Vec<i32>, target: i32) -> i32 {
    use std::cmp::Ordering;
    let (mut left, mut right) = (0, (nums.len() - 1) as i32);
    while left <= right {
        let mid = left + (right - left) / 2;
        match nums[mid as usize].cmp(&target) {
            Ordering::Greater => right = mid - 1,
            Ordering::Less => left = mid + 1,
            Ordering::Equal => return mid,
        }
    }
    left
}

File: Go/search_insert_position/search_insert_position.go
package search_insert_position

func searchInsert(nums []int, target int) int {
	start, end := 0, len(nums)-1

	for start <= end {
		middle := (end-start)/2 + start
		if nums[middle] == target {
			return middle
		}

		if nums[middle] < target {
			start = middle + 1
		} else {
			end = middle - 1
		}
	}

	return start
}
